### PR TITLE
[Unit-test]To avoid regression in Server printing to PDF output format

### PR DESCRIPTION
**Pull request by rldhont**
## Description
In QGIS 3.4, Selection can be printed in Image output and not in PDF or SVG output.

A fix has been done https://github.com/qgis/QGIS/commit/2752f8348a5060dae3834ca40f2645e92ec4b18a to fix inconsistent use of layout render context flags, and draw selection is activated with a flag.

This PR adds unit-tests to avoid regression in QGIS Server context.

The fix inconsistent use of layout render context flags https://github.com/qgis/QGIS/commit/2752f8348a5060dae3834ca40f2645e92ec4b18a will be backported with unit-tests.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/core/layout/qgslayoutexporter.cpp [hunk 1]
```diff
@@ -491,6 +491,7 @@ QgsLayoutExporter::ExportResult QgsLayoutExporter::exportToPdf( const QString &f
   LayoutContextSettingsRestorer contextRestorer( mLayout );
   ( void )contextRestorer;
   mLayout->renderContext().setDpi( settings.dpi );
+  mLayout->renderContext().setFlags( settings.flags );
```
Comment (nyalldawson) [inline src/core/layout/qgslayoutexporter.cpp]: Look two lines below all your changes ;)

Comment (rldhont) [reply src/core/layout/qgslayoutexporter.cpp]: Ok, so the backport has not been yet done ?

Comment (nyalldawson) [reply src/core/layout/qgslayoutexporter.cpp]: I'm unsure - I don't think this was introduced to fix the selection issue, it was introduced to fix font rendering issues.

BTW - it IS possible to test pdf/svg outputs, test_qgslayoutexporter.py has many pdf/svg checks.

Comment (rldhont) [reply src/core/layout/qgslayoutexporter.cpp]: I have found your fix https://github.com/qgis/QGIS/commit/2752f8348a5060dae3834ca40f2645e92ec4b18a it is in master and in release-3_6.

I will update this PR to add some unit tests and I will propose a backport for release-3_4.

**Comment by rldhont** [comment]
@elpaso @pblottiere can you take a look at theses new unit-tests ?

**Review by elpaso — APPROVED** [review]

**Comment by rldhont** [comment]
Thanks  @elpaso

**Pull request merged**